### PR TITLE
Add protocol tests for String & Enum HTTP payloads

### PR DESCRIPTION
**Pull request by rcoh**
*Description of changes:* There are no protocol tests for Strings or enum-marked-strings for restJson.

*Tests*: The Rust SDK passes these tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/restJson1/http-string-payload.smithy [hunk 1]
```diff
@@ -0,0 +1,45 @@
+namespace aws.protocoltests.restjson
```
Comment (kstich) [inline smithy-aws-protocol-tests/model/restJson1/http-string-payload.smithy:1]: Should have a `$version: "1.0"` at the top

File: smithy-aws-protocol-tests/model/restJson1/http-string-payload.smithy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use smithy.test#httpRequestTests
```
Comment (kstich) [inline smithy-aws-protocol-tests/model/restJson1/http-string-payload.smithy:2]: Usually an empty line between the `namespace` and `use` statements.

**Review by kstich — COMMENTED** [review]
Thanks for adding these!

Would you be willing to add a couple tests for responses? No worries if not, we can add them after as well.

**Comment by rcoh** [comment]
yeah I have them locally actually, let me copy them over

**Pull request merged**